### PR TITLE
Remove some unnecessary indirection in tests

### PR DESCRIPTION
**Pull request by lencioni**
I was thinking about making a change to waypoint, but I found the tests
to be unnecessarily difficult to follow, so I decided to first reduce
some of the indirection and nesting. I think there is more that could be
done here, but this seems like a pretty good start.

### CODE REVIEW

**Comment by lencioni** [comment]
This is probably easiest reviewed with the `?w=1` query string param.

File: spec/waypoint_spec.js [hunk 1]
```diff
@@ -90,9 +85,9 @@ describe('<Waypoint>', function() {
       this.scrollable = this.parentComponent;
     });
 
-    it('does not log to the console', () => {
-      expect(console.log) // eslint-disable-line no-console
-        .not.toHaveBeenCalled();
+    it('does not log to the console u', () => {
```
Comment (trotzig) [inline spec/waypoint_spec.js]: did you add the "u" at the end by mistake?

Comment (lencioni) [inline spec/waypoint_spec.js]: oops!

**Comment by lencioni** [comment]
@trotzig Yes, that's right. I'll update the commit message

**Pull request merged**